### PR TITLE
Supporting Customer managed key 

### PR DESCRIPTION
**Pull request by ghost**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public SoftwareUpdateConfiguration CreateSoftwareUpdateConfiguration(string reso
 
                 var sucParameters = new Sdk.SoftwareUpdateConfiguration()
                 {
-                    ScheduleInfo = new Sdk.ScheduleProperties()
+                    ScheduleInfo = new Sdk.SUCScheduleProperties()
```
Comment (dingmeng-xue) [inline src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs:82]: What's the reason of this change?

Comment (sharma224) [reply src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs:82]: ScheduleProperties was updated to SUCScheduleProperties in the swagger definition for Software Update Configuration schedule properties, hence this change

**Comment by dingmeng-xue** [comment]
We can see below changes on `Get-AzAutomationSoftwareUpdateMachineRun` and `Get-AzAutomationSoftwareUpdateRun`

The type of property 'Tasks' of type 'Microsoft.Azure.Commands.Automation.Model.UpdateManagement.SoftwareUpdateRun' has changed from 'Microsoft.Azure.Management.Automation.Models.SoftareUpdateConfigurationRunTasks' to 'Microsoft.Azure.Management.Automation.Models.SoftwareUpdateConfigurationRunTasks'.

**Comment by sharma224** [comment]
> We can see below changes on `Get-AzAutomationSoftwareUpdateMachineRun` and `Get-AzAutomationSoftwareUpdateRun`
> 
> The type of property 'Tasks' of type 'Microsoft.Azure.Commands.Automation.Model.UpdateManagement.SoftwareUpdateRun' has changed from 'Microsoft.Azure.Management.Automation.Models.SoftareUpdateConfigurationRunTasks' to 'Microsoft.Azure.Management.Automation.Models.SoftwareUpdateConfigurationRunTasks'.
> 
> Please confirm that.

Yes, a spell check was made in the swagger for the definition of 'SoftareUpdateConfigurationRunTasks' to 'Soft**w**areUpdateConfigurationRunTasks' and the same is passed on to the SDK and then here.

File: src/Automation/Automation/Cmdlet/NewAzureAutomationAccount.cs [hunk 2]
```diff
@@ -87,14 +97,64 @@ public IAutomationPSClient AutomationClient
...
+            bool enableAMK = false;
+            if (AutomationServicesEncryption.IsPresent)
+            {
+                enableAMK = true;
+            }
+            bool enableCMK = false;
+            if (ParameterSetName == KeyVaultEncryptionParameterSet)
+            {
+                enableCMK = true;
+            }
```
Comment (BethanyZhou) [inline src/Automation/Automation/Cmdlet/NewAzureAutomationAccount.cs:153]:
```suggestion
            bool addSystemId = AssignSystemIdentity.IsPresent;
            bool enableAMK = AutomationServicesEncryption.IsPresent;
            bool enableCMK = (ParameterSetName == KeyVaultEncryptionParameterSet);
```

File: src/Automation/Automation/Cmdlet/SetAzureAutomationAccount.cs [hunk 3]
```diff
@@ -80,13 +90,62 @@ public IAutomationPSClient AutomationClient
...
+            bool enableAMK = false;
+            if (AutomationServicesEncryption.IsPresent)
+            {
+                enableAMK = true;
+            }
+            bool enableCMK = false;
+            if (ParameterSetName == KeyVaultEncryptionParameterSet)
+            {
+                enableCMK = true;
+            }
```
Comment (BethanyZhou) [inline src/Automation/Automation/Cmdlet/SetAzureAutomationAccount.cs:146]:
```suggestion
            bool addSystemId = AssignSystemIdentity.IsPresent;
            bool enableAMK = AutomationServicesEncryption.IsPresent;
            bool enableCMK =  (ParameterSetName == KeyVaultEncryptionParameterSet);
```

File: src/Automation/Automation/help/Set-AzAutomationAccount.md [hunk 4]
```diff
@@ -57,6 +85,62 @@ Accept pipeline input: False
...
+Aliases:
+Required: True
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -KeyVaultEncryption
+Whether to set Automation Account encryption keySource to Microsoft.KeyVault or not. If you specify KeyName, KeyVersion and KeyVaultUri, Automation Account Encryption KeySource will also be set to Microsoft.KeyVault weather this parameter is set or not.
```
Comment (BethanyZhou) [inline src/Automation/Automation/help/Set-AzAutomationAccount.md:103]:
```suggestion
Specify whether set Automation Account encryption keySource to Microsoft.KeyVault or not. If you specify KeyName, KeyVersion and KeyVaultUri, Automation Account Encryption KeySource will also be set to Microsoft.KeyVault whether this parameter is set or not.
```

File: src/Automation/Automation/help/Set-AzAutomationAccount.md [hunk 5]
```diff
@@ -42,6 +42,34 @@ This command changes the plan to Basic for the Automation account named Automati
...
+Aliases:
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -AutomationServicesEncryption
+Whether to set Automation Account Encryption KeySource to Microsoft.AutomationServices or not.
```
Comment (BethanyZhou) [inline src/Automation/Automation/help/Set-AzAutomationAccount.md:60]:
```suggestion
Specify whether set Automation Account Encryption KeySource to Microsoft.AutomationServices or not.
```

File: src/Automation/Automation/Cmdlet/NewAzureAutomationAccount.cs [hunk 6]
```diff
@@ -87,14 +97,64 @@ public IAutomationPSClient AutomationClient
         [Alias("Tag")]
         public IDictionary Tags { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Generate and assign a new System Identity for this automation account")]
+        public SwitchParameter AssignSystemIdentity { get; set; }
+
+        [Parameter(HelpMessage = "Whether to set Automation Account KeySource to Microsoft.Automation or not.",
```
Comment (BethanyZhou) [inline src/Automation/Automation/Cmdlet/NewAzureAutomationAccount.cs:105]:
```suggestion
        [Parameter(HelpMessage = "Specify whether set Automation Account KeySource to Microsoft.Automation or not.",
```

File: src/Automation/Automation/Cmdlet/NewAzureAutomationAccount.cs [hunk 7]
```diff
@@ -87,14 +97,64 @@ public IAutomationPSClient AutomationClient
...
+            Mandatory = false,
+            ParameterSetName = AutomationServicesEncryptionParameterSet)]
+        public SwitchParameter AutomationServicesEncryption { get; set; }
+
+        [Parameter(HelpMessage = "Whether to set Automation Account KeySource to Microsoft.KeyVault(enable CMK) or not.",
```
Comment (BethanyZhou) [inline src/Automation/Automation/Cmdlet/NewAzureAutomationAccount.cs:110]:
```suggestion
        [Parameter(HelpMessage = "Specifiy whether set Automation Account KeySource to Microsoft.KeyVault (enable CMK) or not.",
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs [hunk 8]
```diff
@@ -57,6 +57,6 @@ internal SoftwareUpdateRun(string resourceGroupName, string automationAccountNam
 
         public SoftwareUpdateRunStatus Status { get; set; }
 
-        public SoftareUpdateConfigurationRunTasks Tasks { get; set; }
+        public SoftwareUpdateConfigurationRunTasks Tasks { get; set; }
```
Comment (BethanyZhou) [inline src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs:60]: why we need change SoftareUpdateConfigurationRunTasks to SoftwareUpdateConfigurationRunTasks, this change will introduce breaking changes.

Comment (BethanyZhou) [reply src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs:60]: If this change is necessary
- Please suppress the detection of breaking change in azure-powershell/tools/StaticAnalysis/Exceptions/Az.Automation/BreakingChangeIssues.csv. 
- Please wait for our major release (about middle of May), because breaking changes are only allowed on major release.

File: src/Automation/Automation/ChangeLog.md [hunk 9]
```diff
@@ -18,11 +18,12 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
-* Fixed issue that disables the schedule for update deployment if schedule was re-created with same name
```
[the remark this answers is not in the document]
Comment (sharma224) [reply src/Automation/Automation/ChangeLog.md:21]: Changed it

File: src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs (hunk 8 above)
Comment (sharma224) [reply src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs:60]: Yes, this is necessary as a spelling correction was made in the swagger. Have added the BreakingChangeIssues.csv file.

Comment (dingmeng-xue) [reply src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs:60]: We cannot measure the usage of `SoftareUpdateConfigurationRunTasks` but we assume the impact is not small because that parameter was added 2 years ago. `Get-AzAutomationSoftwareUpdateRun` has been called 240K times in the past 30 days and `Get-AzAutomationSoftwareUpdateMachineRun` has been called 73K times. Given the impact and importancy of Az.Autoamtion, we recommend that you add this feature in the mid of May.

Comment (sharma224) [reply src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateRun.cs]: Created 'SoftareUpdateConfigurationRunTasks' based on 'SoftwareUpdateConfigurationRunTasks' to avoid the breaking change temporarily

File: tools/StaticAnalysis/Exceptions/Az.Automation/BreakingChangeIssues.csv [hunk 10]
```diff
@@ -0,0 +1,3 @@
+AssemblyFileName,ClassName,Target,Severity,ProblemId,Description,Remediation
```
Comment (BethanyZhou) [inline tools/StaticAnalysis/Exceptions/Az.Automation/BreakingChangeIssues.csv:1]: Please remove the breakingchangeissues.csv bc no breaking changes are introduced.

Comment (sharma224) [reply tools/StaticAnalysis/Exceptions/Az.Automation/BreakingChangeIssues.csv]: Removed BreakingChangeIssues.csv

**Review by BethanyZhou — APPROVED** [review]
LGTM

File: src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.cs [hunk 11]
```diff
@@ -107,7 +107,7 @@ public void TestCreateAndGetSoftwareUpdateConfigurationWithPrePost()
             TestRunner.RunTestScript("Test-CreateAndGetSoftwareUpdateConfigurationWithPrePost");
         }
 
-        [Fact]
+        [Fact(Skip = "Temporarily skipping")]
```
Comment (BethanyZhou) [inline src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.cs:110]: I think this test shouldn't skip here, should it?

Comment (sharma224) [reply src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.cs]: Did it temporarily, have updated the recording for this test.

**Review by BethanyZhou — APPROVED** [review]
LGTM
